.github/workflows/ci.yml

```diff
@@ -0,0 +1,43 @@
+name: Cross-Platform CI
+
+on:
```

Comment:
Do we want this to run for any change? I think it's more beneficial to focus on actual code changes. See the `paths` suggestion.